rollup/da_syncer/da_queue.go

```diff
@@ -0,0 +1,67 @@
+package da_syncer
+
+import (
+	"context"
+	"errors"
+
+	"github.com/scroll-tech/go-ethereum/rollup/da_syncer/da"
+)
+
+// DAQueue is a pipeline stage that reads DA entries from a DataSource and provides them to the next stage.
+type DAQueue struct {
+	l1height          uint64
+	dataSourceFactory *DataSourceFactory
+	dataSource        DataSource
+	da                da.Entries
+}
+
+func NewDAQueue(l1height uint64, dataSourceFactory *DataSourceFactory) *DAQueue {
+	return &DAQueue{
+		l1height:          l1height,
+		dataSourceFactory: dataSourceFactory,
+		dataSource:        nil,
+		da:                make(da.Entries, 0),
+	}
+}
+
+func (dq *DAQueue) NextDA(ctx context.Context) (da.Entry, error) {
+	for len(dq.da) == 0 {
+		err := dq.getNextData(ctx)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
should we time.sleep here?

Reply:
if it is an error that can be handled, it will be handled in `syncing_pipeline` and next step will be started with some delay depending on which error is it

Reply:
I don't think all the cases are covered.
for example, if we `ds.l1Client.GetLatestFinalizedBlockNumber()` or `ds.l1Client.FetchRollupEventsInRange(ds.l1height, to)` error
we will keep querying them infinitely

Reply:
Actually, this is a very good point. But I'm not sure we will keep trying infinitely. Like @NazariiDenha said, we should handle such error at the event loop in `SyncingPipeline` but we currently don't. This would mean not querying them infinitely but exiting the pipeline if any such error occurs. This is bad. We should introduce a temporary error or something that we check for in the event loop and backoff + retry (max backoff is currently configured at 10s).

Reply:
Done in 8f1f6b725ed8133f3560805ac6227d87c6305414